source/common/router/rds_impl.cc

```diff
@@ -116,10 +117,23 @@ void RdsRouteConfigSubscription::onConfigUpdate(
     config_info_ = {new_hash, version_info};
     route_config_proto_ = route_config;
     stats_.config_reload_.inc();
-    ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
-              new_hash);
-    for (auto* provider : route_config_providers_) {
-      provider->onConfigUpdate();
+
+    if (!route_config_proto_.has_vhds()) {
+      ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
+                new_hash);
+
+      last_update_details_ = std::make_unique<RdsConfigUpdateDetails>(route_config_proto_,
```

Comment:
Would having VHDS and RDS details distinct vs. merged provide any benefit? I'm thinking specifically about the idea we once had of allowing the config dump to be used to reconstruct the original configuration. It's a bit hokey having to plumb route config info down to VHDS to provide config dump info, but it works.

Reply:
It's a bit more involved: ```VhdsSubscription``` uses RouteConfiguration it receives from the parent ```RdRdsRouteConfigSubscription``` to build a complete RouteConfiguration (rds + vhds). ```RdsRouteConfigProviderImpl``` uses the complete RouteConfiguration then to update worker threads.

I like the idea of merging RDS and VHDS update details into one class. We can then pass an instance of it into ```RouteConfigProvider.onConfigUpdate()``` (https://github.com/envoyproxy/envoy/blob/master/source/common/router/rds_impl.h#L155), instead of ```RdsRouteConfigProvider``` reaching into ```RdRdsRouteConfigSubscription```. In other words to use it both for updating RouteConfig in worker threads and for generation of config dumps. 

What do you think?

Reply:
Sounds like a nice potential cleanup. We could punt on it to a followup PR though in the interest of reducing churn or do it here as you prefer.

Reply:
I think I'd rather do it here, tbh.

Reply:
Done, see https://github.com/envoyproxy/envoy/pull/6438/files#diff-7ccfbe677293512b58aac30fa4e8c26a.